wallet/wallet-cli/src/main.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod commands;
+mod config;
+mod errors;
+mod helpers;
+mod output;
+mod repl;
+mod wallet_init;
+
+use std::sync::Arc;
+
+use clap::Parser;
+use dialoguer::theme::ColorfulTheme;
+use errors::WalletCliError;
+use node_comm::make_rpc_client;
+use output::OutputContext;
+use wallet::{Wallet, WalletError};
+
+type DefWallet = Wallet<wallet_storage::DefaultBackend>;
+
+async fn run(output: &OutputContext) -> Result<(), WalletCliError> {
+    logging::init_logging::<&std::path::Path>(None);
+
+    let args = config::WalletCliArgs::parse();
+    let config = config::WalletCliConfig::from_args(args)?;
+    let chain_config = Arc::new(common::chain::config::Builder::new(config.chain_type).build());
+
+    let db = wallet::wallet::open_or_create_wallet_file(&config.wallet_file)
+        .map_err(WalletCliError::WalletError)?;
+
+    let theme = ColorfulTheme::default();
+
+    let wallet = match Wallet::load_wallet(Arc::clone(&chain_config), Arc::clone(&db)) {
+        Ok(wallet) => wallet,
+        Err(WalletError::WalletNotInitialized) => {
+            wallet_init::new_wallet(output, chain_config, db, &theme)?
+        }
```

Comment:
We have to be careful and not overwrite the user wallet.

If you see how Monero does this, usually you can open a wallet at a specific path, and if the wallet isn't found, the user is asked to create a wallet at that location.